pkg/apis/batch/validation/validation.go

```diff
@@ -124,11 +124,22 @@ func validateJobSpec(spec *batch.JobSpec, fldPath *field.Path) field.ErrorList {
 	}
 
 	allErrs = append(allErrs, apivalidation.ValidatePodTemplateSpec(&spec.Template, fldPath.Child("template"))...)
```

Comment:
Doing fixup of the returned errors is fragile. If we want to do context-specific validation like this, we should inform the pod template validation function of the allowed values, though the amount of plumbing changes required to do that is significant

Reply:
hi @liggitt  Do you have any better suggestions? I am willing to fix this problem as you think. At present, I personally prefer this modification, because the amount of modification is small, and modifying the return value in this place can cover all job / cronjob situations, because this function is solely for verifying JobSpec and will not affect other original resources Influence the logic.

Reply:
The change here is small but fragile, making assumptions that all errors reported on that field will be related to the allowed values. We don't have a good way to do different validation of complex deeply nested shared structs today while sharing lots of common validation. I don't think a one-off fix here is worth it for the better message.

Reply:
Ok. That can only be done first, thank you for your code review